python/GafferTest/CompoundNumericPlugTest.py

```diff
@@ -175,6 +175,21 @@ def testDynamicSerialisation( self ) :
 
 		self.assertEqual( s["n"]["p"].getValue(), imath.V3f( 1, 2, 3 ) )
 
+	def testDynamicSerialisationNonDefaultInterpretation( self ):
+
+		s = Gaffer.ScriptNode()
+		n = Gaffer.Node()
+		n["p"] = Gaffer.V3fPlug( flags=Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, interpretation = IECore.GeometricData.Interpretation.Vector )
+		n["p"].setValue( imath.V3f( 1, 2, 3 ) )
+		s["n"] = n
+
+		ss = s.serialise()
+
+		s = Gaffer.ScriptNode()
+		s.execute( ss )
+
+		self.assertEqual( s["n"]["p"].getValue(), imath.V3f( 1, 2, 3 ) )
```

Comment:
We have plenty of test coverage for the serialisation of _values_ already - I think what we need here is to assert that `s["n"]["p"].interpretation()` is what we expect.

Reply:
The bug causes an Exception to be raised from s.execute( ss ) so the assertEqual line is actually redundant, I can change it to test the interpretations match if that gives extra coverage but its not really what is being tested here.